barretenberg/cpp/src/barretenberg/vm2/constraining/relations/get_env_var.test.cpp

```diff
@@ -0,0 +1,787 @@
+          { C::execution_envvar_pi_row_idx, 0 }, // none
+          { C::execution_is_address, 0 },
+          { C::execution_is_sender, 0 },
+          { C::execution_is_transactionfee, 0 },
+          { C::execution_is_isstaticcall, 0 },
+          { C::execution_is_l2gasleft, 1 },
+          { C::execution_is_dagasleft, 0 },
+          // Do it! No prior error, and enum is valid
+          { C::execution_sel_should_get_env_var, 1 },
+          // output
+          { C::execution_register_0_, l2_gas_limit - l2_gas_used },
+          // nothing from pis
+          { C::execution_value_from_pi, 0 } },
+        // daGasLeft
+        { { C::execution_sel, 1 },
+          { C::execution_sel_get_env_var, 1 },
+          { C::execution_da_gas_limit, da_gas_limit },
+          { C::execution_da_gas_used, da_gas_used },
+          { C::execution_rop_1_, static_cast<uint8_t>(EnvironmentVariable::DAGASLEFT) },
+          { C::execution_sel_should_resolve_address, 1 },
+          { C::execution_sel_addressing_error, 0 },
+          // from precomputed table
+          { C::execution_opcode_error, 0 }, // valid enum
+          { C::execution_sel_envvar_pi_lookup_col0, 0 },
+          { C::execution_sel_envvar_pi_lookup_col1, 0 },
+          { C::execution_envvar_pi_row_idx, 0 }, // none
+          { C::execution_is_address, 0 },
+          { C::execution_is_sender, 0 },
+          { C::execution_is_transactionfee, 0 },
+          { C::execution_is_isstaticcall, 0 },
+          { C::execution_is_l2gasleft, 0 },
+          { C::execution_is_dagasleft, 1 },
+          // Do it! No prior error, and enum is valid
+          { C::execution_sel_should_get_env_var, 1 },
+          // output
+          { C::execution_register_0_, da_gas_limit - da_gas_used },
+          // nothing from pis
+          { C::execution_value_from_pi, 0 } },
+        // invalid enum
+        { { C::execution_sel, 1 },
+          { C::execution_sel_get_env_var, 1 },
+          { C::execution_rop_1_, 42 },
+          { C::execution_sel_should_resolve_address, 1 },
+          { C::execution_sel_addressing_error, 0 },
+          // from precomputed table
+          { C::execution_opcode_error, 1 }, // invalid enum!
+          { C::execution_sel_envvar_pi_lookup_col0, 0 },
+          { C::execution_sel_envvar_pi_lookup_col1, 0 },
+          { C::execution_envvar_pi_row_idx, 0 }, // none
+          { C::execution_is_address, 0 },
+          { C::execution_is_sender, 0 },
+          { C::execution_is_transactionfee, 0 },
+          { C::execution_is_isstaticcall, 0 },
+          { C::execution_is_l2gasleft, 0 },
+          { C::execution_is_dagasleft, 0 },
+          // ERROR! Invalid enum
+          { C::execution_sel_should_get_env_var, 0 },
+          // output
+          { C::execution_register_0_, 0 }, // no output on error
+          // nothing from pis
+          { C::execution_value_from_pi, 0 } },
+    });
+
+    tracegen::PublicInputsTraceBuilder public_inputs_builder;
```

Comment:
In tests, prefer doing `using tracegen::PublicInputsTraceBuilder` globally. Also for any other `tracegen::` or `simulation::`